Edit mode and filters

MishaBlin opened the pull request.

Zaurall committed:
Add filters form, toggle switch

Zaurall committed:
Configured layout, completed filters, adding forms

Zaurall committed:
Added stores. Can save forms, when changing pages

Zaurall committed:
Fixed error with adding player

Zaurall committed:
Changed name of function

MishaBlin committed:
Edit mode sample.

MishaBlin committed:
Merge branch 'edit-mode-and-filters'

MishaBlin committed:
Comment checkbox out of players for testing

Zaurall committed:
Fixed layout, complete filters, sort

MishaBlin committed:
Edit mode test for matches

MishaBlin committed:
Edit mode test for matches

MishaBlin committed:
Edit mode fix

MishaBlin committed:
Edit mode fix and warnings suppressed

Zaurall committed:
Complete layout, filters and sort, fix bugs

KekStroke commented on src/routes/tournaments/+page.svelte:
```diff
@@ -4,21 +4,89 @@
 	import AddTournamentForm from '$lib/components/AddTournamentForm.svelte';
-	import Separator from '$lib/components/decorations/Separator.svelte';
 	import TournamentList from '$lib/components/TournamentList.svelte';
+	// import ToggleCheckboxButton from '$lib/components/base/ToggleCheckboxButton.svelte';
```
Коммент удалить

Zaurall replied:
Done

KekStroke commented on src/routes/tournaments/+page.svelte:
```diff
@@ -4,21 +4,89 @@
 
-{#if isLeader}
-	<AddTournamentForm on:error={handleError} on:update={() => handleInsert()} />
+<div class="page">
```
Убрать этот wrapper

Zaurall replied:
done

KekStroke commented on src/routes/tournaments/+page.svelte:
```diff
@@ -4,21 +4,89 @@
 
-	<Separator />
-{/if}
+	<div class="wrapper">
```
Придумать более осознанное название класса

Zaurall replied:
done

KekStroke commented on src/routes/tournaments/+page.svelte:
```diff
@@ -4,21 +4,89 @@
+			<div class="form">
+				<AddTournamentForm on:error={handleError} on:update={() => handleInsert()} />
+			</div>
+		{:else if $userToken}
```
Вынести логику с проверкой токена общий layout

Zaurall replied:
done

KekStroke commented on src/routes/tournaments/+page.svelte:
```diff
@@ -4,21 +4,89 @@
+<style>
+	.page {
+		padding: 0 5%;
+	}
```
убрать

Zaurall replied:
done

KekStroke commented on src/routes/tournaments/+page.svelte:
```diff
@@ -4,21 +4,89 @@
+		padding: 0 5%;
+	}
+	.wrapper {
+		height: 600px;
```
Не писать в px

Zaurall replied:
done

KekStroke commented on src/routes/players/+page.svelte:
```diff
@@ -10,12 +11,78 @@
+		padding: 0 5%;
+	}
+	.wrapper {
+		height: 600px;
```
Поменять px

Zaurall replied:
done

KekStroke commented on src/routes/players/+page.svelte:
```diff
@@ -10,12 +11,78 @@
+			<div class="form">
+				<AddPlayerForm on:error={handleError} on:update={() => handleInsert()} />
+			</div>
+		{:else if $userToken}
```
вы нести в layout

Zaurall replied:
done

KekStroke commented on src/lib/components/icons/DescendingIcon.svelte:
```diff
@@ -0,0 +1,25 @@
+	export let disabled = false;
+</script>
+
+<svg width="60" height="60" viewBox="0 0 60 60" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Сделать width и height через пропы

Zaurall replied:
done

KekStroke commented on src/lib/stores.ts:
```diff
@@ -1,6 +1,15 @@
+	addPlayerFormStore,
+	addTournamentFormStore,
+	addMatchFormStore,
+} from '../lib/types/types';
```
```suggestion
} from '$lib/types/types';
```

Zaurall replied:
done

KekStroke commented on src/lib/stores.ts:
```diff
@@ -15,3 +24,49 @@ export const userToken: Writable<string> = writable('');
 export const errors: Writable<Error[]> = writable([]);
+
+const SORT_FILTER_PLAYER_FORM: sortFilterPlayerFormStore = {
+	name: '',
+	telegramAlias: '',
+	minRating: '',
+	maxRating: '',
+	descending: true,
+	sortBy: 'rating',
+};
+const SORT_FILTER_TOURNAMENT_FORM: sortFilterTournamentFormStore = {
+	title: '',
+	minParticipants: '',
+	maxParticipants: '',
+	startDateString: '',
+	endDateString: '',
+	descending: true,
+	sortBy: 'date',
+};
+const SORT_FILTER_MATCH_FORM: sortFilterMatchFormStore = {
+	name: '',
+	score: '',
+	minDateString: '',
+	maxDateString: '',
+	descending: true,
+	sortBy: 'date',
+};
+const ADD_PLAYER_FORM: addPlayerFormStore = { name: '', telegramAlias: '', initialRating: 100 };
+const ADD_TOURNAMENT_FORM: addTournamentFormStore = {
+	title: '',
+	startDateString: convertDateToStringDash(new Date()),
+	endDateString: convertDateToStringDash(new Date()),
+};
+const ADD_MATCH_FORM: addMatchFormStore = {
+	firstPlayerName: '',
+	secondPlayerName: '',
+	tournamentTitle: '',
+	firstPlayerScore: 0,
+	secondPlayerScore: 0,
+};
```
вынести в отдельный файл

Zaurall replied:
done

KekStroke commented on src/lib/types/types.ts:
```diff
@@ -41,4 +41,52 @@ export interface Tournaments {
+	tournamentTitle: string;
+	firstPlayerScore: number;
+	secondPlayerScore: number;
+}
```
Вынести в отдельный файл storeTypes.ts или что-то типо такого

KekStroke replied:
Или formStores

Zaurall replied:
done

KekStroke commented on src/lib/components/icons/AscendingIcon.svelte:
```diff
@@ -0,0 +1,25 @@
+	export let disabled = true;
+</script>
+
+<svg width="60" height="60" viewBox="0 0 60 60" fill="none" xmlns="http://www.w3.org/2000/svg">
```
width, height 100%

Zaurall replied:
done

KekStroke commented on src/lib/components/base/ToggleCheckboxButton.svelte:
```diff
@@ -0,0 +1,74 @@
+
+	.toggle-switch {
+		display: inline-block;
+		/* background: var(--main-color); */
```
Почистить комменты

Zaurall replied:
done

KekStroke commented on src/lib/components/TournamentList.svelte:
```diff
@@ -148,10 +167,10 @@
 	.table-header.not-leader {
 		grid-template-columns: 1fr 5.7em 5.7em 5.7em;
-	}
+	} */
```
Это комменты или не комменты? Почистить

Zaurall replied:
done

KekStroke commented on src/lib/components/TournamentList.svelte:
```diff
@@ -102,13 +121,13 @@
 			</section>
 		</Pagination>
 	{:else}
-		<p class="details">Oops! That page doesn't exist or it is private</p>
+		<p class="details">Oops! There is not a single entity satisfying the query</p>
```
Оставить как было и перенести в layout

Zaurall replied:
done

KekStroke commented on src/lib/components/SortFilterTournamentForm.svelte:
```diff
@@ -0,0 +1,243 @@
+	};
+
+	const saveForm = function () {
+		console.log(endDateString);
```
```suggestion
```

Zaurall replied:
done

KekStroke commented on src/lib/components/SortFilterTournamentForm.svelte:
```diff
@@ -0,0 +1,243 @@
+	const saveForm = function () {
+		console.log(endDateString);
+		const sortby: 'date' | 'kf' | 'numberOfPlayers' =
+			sortBy === 'date' ? 'date' : sortBy === 'kf' ? 'kf' : 'numberOfPlayers';
```
Убрать, прописать тип в начале

Zaurall replied:
done

KekStroke commented on src/lib/components/SortFilterTournamentForm.svelte:
```diff
@@ -0,0 +1,243 @@
+	}
+	.sorting-order input {
+		position: fixed;
+		opacity: 0;
```
display: hidden

Zaurall replied:
done

KekStroke commented on src/lib/components/SortFilterPlayerForm.svelte:
```diff
@@ -0,0 +1,219 @@
+	}
+	.sorting-order input {
+		position: fixed;
+		opacity: 0;
```
display: hidden

Zaurall replied:
done

KekStroke commented on src/lib/components/SortFilterPlayerForm.svelte:
```diff
@@ -0,0 +1,219 @@
+<script lang="ts">
+	// import { enhance } from '$app/forms';
+	import { get } from 'svelte/store';
```
get не нужен, используй везде $

Zaurall replied:
done

KekStroke commented on src/lib/components/SortFilterPlayerForm.svelte:
Переписать всю работу со сторами через $-синтаксис

Zaurall replied:
done

KekStroke commented on src/lib/components/SortFilterTournamentForm.svelte:
Переписать всю работу со сторами через $-синтаксис

Zaurall replied:
done

KekStroke commented on src/lib/components/SortFilterMatchForm.svelte:
```diff
@@ -0,0 +1,223 @@
+	}
+	.sorting-order input {
+		position: fixed;
+		opacity: 0;
```
display:hidden

Zaurall replied:
done

KekStroke commented on src/lib/components/SortFilterMatchForm.svelte:
Переписать всю работу со сторами через $-синтаксис

Zaurall replied:
done

KekStroke commented on src/lib/components/PlayersList.svelte:
```diff
@@ -98,7 +116,7 @@
 			</section>
 		</Pagination>
 	{:else}
-		<p class="details">Oops! That page doesn't exist or it is private</p>
+		<p class="details">Oops! There is not a single entity satisfying the query</p>
```
Оставить как было, вынести в отдельный layout

KekStroke commented on src/lib/components/AddTournamentForm.svelte:
```diff
@@ -43,52 +45,66 @@
+			title: title,
+			startDateString: startDateString,
+			endDateString: endDateString,
+		});
```
Через $ синтаксис

KekStroke commented on src/lib/components/AddTournamentForm.svelte:
Переписать всю работу со сторами через $-синтаксис

Zaurall committed:
Add new layout for tournaments and players pages

KekStroke commented on src/lib/components/AddMatchForm.svelte:
Переписать всю работу со сторами через $-синтаксис

MishaBlin committed:
Deleted unnecessary binds.
Hover for buttons in matches list.

KekStroke commented on src/lib/components/MatchesList.svelte:
```diff
@@ -50,7 +68,7 @@
 			dispatch('error', error);
 		});
-
+		console.log(data);
```
Убрать логи

MishaBlin committed:
Merge branch 'edit-mode-and-filters' of https://github.com/InnoTableTennis/frontend into edit-mode-and-filters

MishaBlin committed:
Format

KekStroke commented on src/lib/components/MatchesList.svelte:
```diff
@@ -70,51 +88,62 @@
+								{match.firstPlayerName}
+								<span class="rating">
+									{#if match.firstPlayerRatingDelta}
+										({match.firstPlayerRatingBefore})
```
Показывать, даже если delta == 0

KekStroke replied:
Это баг, надо пофиксить

KekStroke commented on src/lib/components/base/DropdownInput.svelte:
```diff
@@ -21,7 +21,8 @@
 	let unorderedList: HTMLUListElement;
-	let inputVal = '';
+
+	export let inputVal = '';
```
не должен быть пропом

KekStroke commented on src/routes/+page.svelte:
```diff
@@ -24,17 +32,118 @@
-			on:error={handleError}
-			on:update={() => handleInsert()}
-		/>
+<div class="page">
```
Убрать этот враппер

Zaurall replied:
done

KekStroke commented on src/routes/+page.svelte:
```diff
@@ -24,17 +32,118 @@
-		<Separator />
-	{/await}
-{/if}
+	<div class="wrapper">
```
Переименновать класс

Zaurall replied:
done

MishaBlin committed:
Merge branch 'main' into edit-mode-and-filters

KekStroke commented on src/routes/+page.svelte:
```diff
@@ -24,17 +32,118 @@
+		max-width: 900px;
+	}
+
+	@media (max-width: 1300px) {
+		.page {
+			padding: 0;
+		}
+		.form {
+			margin-right: 2rem;
+		}
+	}
```
Это можно убрать

Zaurall replied:
done

KekStroke commented on src/routes/+page.svelte:
```diff
@@ -4,18 +4,26 @@
+	$: editData;
 	$: isLeader = getRoles($userToken).includes('LEADER');
+	$: isEditing = false;
+	$: mode = 'add';
```
Реактивность не нужна

Zaurall replied:
done

KekStroke commented on src/routes/+page.svelte:
```diff
@@ -4,18 +4,26 @@
+	$: isEditing = false;
+	$: mode = 'add';
+	$: isChoosing = (mode === 'edit' || mode === 'delete') && isEditing;
+	$: chosenId = -1;
```
Реактивность не нужна

Zaurall replied:
done

KekStroke reviewed (changes requested):
Добавить placeholder в фильтер по дате
Фикс фильрра по дате

KekStroke reviewed (changes requested):
Добавить placeholder в фильтер по дате

MishaBlin committed:
Merge branch 'main' of https://github.com/InnoTableTennis/frontend into edit-mode-and-filters

MishaBlin committed:
Colors and theme fix

MishaBlin committed:
Added input components.
Deleted unnecessary reactivity.
Fixed MatchesList button highlighting.

Zaurall committed:
Change of Pagination

Zaurall committed:
Merge branch 'edit-mode-and-filters' of github.com:InnoTableTennis/frontend into edit-mode-and-filters

MishaBlin committed:
SwitchBar and MatchesList buttons styling

MishaBlin committed:
Edit mode to players and tournaments.
Input components enhancement.
Added text input component.
Requests to edit tournaments and players.

MishaBlin committed:
Warnings suppressed

Zaurall committed:
Not completed new design of tables

Zaurall committed:
New desogn of tables

Zaurall committed:
New design of tables

Zaurall committed:
New design of tables (player, tournament)

MishaBlin committed:
Input coponents implementation and fix

MishaBlin committed:
Fixed imports

MishaBlin committed:
Input component template

MishaBlin committed:
Merge branch 'main' of https://github.com/InnoTableTennis/frontend into edit-mode-and-filters

Zaurall committed:
New design

Zaurall committed:
Merge branch 'edit-mode-and-filters' of github.com:InnoTableTennis/frontend into edit-mode-and-filters

Zaurall committed:
Scrollable tables

MishaBlin committed:
Delete functionality + popup

MishaBlin committed:
Merge branch 'edit-mode-and-filters' of https://github.com/InnoTableTennis/frontend into edit-mode-and-filters

MishaBlin committed:
Merge

MishaBlin committed:
Delete functionality + forms fix

Zaurall committed:
New font

Zaurall committed:
Merge branch 'edit-mode-and-filters' of github.com:InnoTableTennis/frontend into edit-mode-and-filters

KekStroke commented on src/lib/components/AddMatchForm.svelte:
```diff
@@ -28,19 +31,19 @@
+			(firstPlayerScore !== 0 || secondPlayerScore !== 0) &&
 			tournamentTitle !== null
 		);
 	}
```
Этот код не будет работать реактивно. Нужно либо переписать объявления по типу
$: firstPlayerName = $AddMatchFormStore.firstPlayerName;
Либо нужно сразу isSubmissionDisabled обновлять по значениям сторов по типу
$: { isSubmissionDisabled != ( $AddmatchFormStore.firstPlayerName & ... )}
Но код дальше написан так, что лучше будет использовать первый вариант

Zaurall replied:
done

KekStroke commented on src/lib/components/AddMatchForm.svelte:
```diff
@@ -28,19 +31,19 @@
+	let secondPlayerName = $AddMatchFormStore.secondPlayerName;
+	let tournamentTitle = $AddMatchFormStore.tournamentTitle;
+	let firstPlayerScore = $AddMatchFormStore.firstPlayerScore;
+	let secondPlayerScore = $AddMatchFormStore.secondPlayerScore;
```
```suggestion
	$: firstPlayerName = $AddMatchFormStore.firstPlayerName;
	$: secondPlayerName = $AddMatchFormStore.secondPlayerName;
	$: tournamentTitle = $AddMatchFormStore.tournamentTitle;
	$: firstPlayerScore = $AddMatchFormStore.firstPlayerScore;
	$: secondPlayerScore = $AddMatchFormStore.secondPlayerScore;
```

Zaurall replied:
done

KekStroke commented on src/lib/components/AddMatchForm.svelte:
```diff
@@ -76,25 +79,42 @@
+			tournamentTitle: tournamentTitle,
+			firstPlayerScore: firstPlayerScore,
+			secondPlayerScore: secondPlayerScore,
+		});
```
```suggestion
		$AddMatchFormStore = {
			firstPlayerName: firstPlayerName,
			secondPlayerName: secondPlayerName,
			tournamentTitle: tournamentTitle,
			firstPlayerScore: firstPlayerScore,
			secondPlayerScore: secondPlayerScore,
		};
```

Zaurall replied:
done

KekStroke reviewed (commented):
Переписать на эффективное использование сторов (ниже написано как). Я дальше не стал проверять. Там тоже самое

Zaurall committed:
Complete new design of forms and tables

KekStroke commented on src/lib/types/types.ts:
```diff
@@ -39,6 +39,7 @@ export interface Tournaments {
 	endDateString: string;
 	finished: boolean;
 	id: number;
+	players: number;
```
playersAmout

Zaurall replied:
to do

MishaBlin replied:
done

Zaurall replied:
done
no changes

KekStroke commented on src/routes/(tournament_player_pages)/tournaments/+page.ts:
Переименновать группу в (private)

Zaurall replied:
done

MishaBlin replied:
done

Zaurall committed:
New private folder

KekStroke commented on src/lib/components/base/pagination/PaginationBar.svelte:
```diff
@@ -77,75 +78,84 @@
+		height: 3rem;
 		margin: 0 auto;
-		margin-bottom: 3em;
+		/* margin-bottom: 3em; */
```
```suggestion
```

MishaBlin replied:
done

Zaurall replied:
done

Zaurall replied:
done

KekStroke commented on src/lib/components/base/inputs/InputTemplate.svelte:
```diff
@@ -0,0 +1,102 @@
+	export let placeholder: string;
+
+	export let defaultValue = '';
+	export let defaultNumValue: number | string = 0;
```
number | ''

Zaurall replied:
done
no changes

KekStroke commented on src/lib/components/base/OrderButton.svelte:
```diff
@@ -0,0 +1,52 @@
+	export let defaultVal = true;
+	export let value = false;
+
+	$: value = defaultVal;
```
```suggestion
```

Zaurall replied:
done

KekStroke commented on src/lib/components/base/OrderButton.svelte:
```diff
@@ -0,0 +1,52 @@
+	import AscendingIcon from '$lib/components/icons/AscendingIcon.svelte';
+	import DescendingIcon from '$lib/components/icons/DescendingIcon.svelte';
+
+	export let defaultVal = true;
+	export let value = false;
```
```suggestion
	export let value = true;
```

Zaurall replied:
done

MishaBlin committed:
Merge branch 'main' into edit-mode-and-filters

MishaBlin committed:
merge

MishaBlin committed:
Merge branch 'edit-mode-and-filters' of https://github.com/InnoTableTennis/frontend into edit-mode-and-filters

KekStroke commented on src/lib/components/TournamentList.svelte:
```diff
@@ -1,13 +1,14 @@
 
 	import { createEventDispatcher } from 'svelte';
-	import { alertPopup } from "$lib/popupHandler";
+	import PlayersIcon from './icons/PlayersIcon.svelte';
```
```suggestion
	import PlayersIcon from '$lib/components/icons/PlayersIcon.svelte';
```

MishaBlin replied:
done

Zaurall replied:
done

MishaBlin committed:
Styles fix + font

Zaurall committed:
Merge branch 'edit-mode-and-filters' of github.com:InnoTableTennis/frontend into edit-mode-and-filters

Zaurall committed:
New routes

MishaBlin committed:
Import changed.
Comment deleted.
PlayersAmount instead of players.

MishaBlin committed:
Format

Zaurall committed:
Form folder and new saving

Zaurall committed:
Merge branch 'edit-mode-and-filters' of github.com:InnoTableTennis/frontend into edit-mode-and-filters

Zaurall committed:
All changes

KekStroke commented on src/lib/components/forms/SortFilterMatchForm.svelte:
```diff
@@ -0,0 +1,175 @@
+		saveForm();
+	};
+
+	function updateValue(event: CustomEvent) {
```
Поменять на updateSortValue или updateRadioGroupValue

Zaurall replied:
done

KekStroke commented on src/lib/components/forms/SortFilterPlayerForm.svelte:
```diff
@@ -0,0 +1,167 @@
+		saveForm();
+	};
+
+	function updateValue(event: CustomEvent) {
```
Поменять название функции

Zaurall replied:
done

KekStroke commented on src/lib/components/forms/SortFilterTournamentForm.svelte:
```diff
@@ -0,0 +1,190 @@
+		saveForm();
+	};
+
+	function updateValue(event: CustomEvent) {
```
Поменять название функции

Zaurall replied:
done

KekStroke commented on src/lib/components/forms/EditMatchForm.svelte:
Почему тут нет функций save и reset?

Zaurall replied:
done

KekStroke commented on src/lib/components/icons/LeftArrow.svelte:
```diff
@@ -1,6 +1,13 @@
-<svg width="16" height="26" viewBox="0 0 16 26" fill="none" xmlns="http://www.w3.org/2000/svg">
+<script lang="ts">
+	export let width = 16;
+	export let height = 26;
```
rem или проценты

Zaurall replied:
done

KekStroke commented on src/lib/components/icons/RightArrow.svelte:
```diff
@@ -1,6 +1,13 @@
-<svg width="16" height="26" viewBox="0 0 16 26" fill="none" xmlns="http://www.w3.org/2000/svg">
+<script lang="ts">
+	export let width = 16;
+	export let height = 26;
```
rem или проценты

Zaurall replied:
done

Zaurall committed:
Pull request resolve

Zaurall committed:
Lint

Zaurall committed:
TS pull resolve
